robot/impl/resource_manager.go

```diff
@@ -1154,6 +1165,12 @@ func (manager *resourceManager) updateResources(
 		}
 		manager.processConfigs[p.ID] = p
 	}
+	for _, c := range conf.OnlyModifiedRevision.Components {
+		manager.updateRevision(c.ResourceName(), c.Revision)
+	}
+	for _, s := range conf.OnlyModifiedRevision.Services {
+		manager.updateRevision(s.ResourceName(), s.Revision)
+	}
```

Comment:
should we do this first?

Reply:
yep we could do this earlier

Reply:
handled much earlier as of https://github.com/viamrobotics/rdk/pull/4206/commits/c5c3c7dd959ab257ef765abf9b1948a5bc9ac72b